### PR TITLE
fix: LEAP-999: Fix E2E tests on LSO

### PR DESCRIPTION
**Pull request by juliosgarbi**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

### Describe the reason for change
The E2E tests have been consistently failing on the LSO environment. Initial investigations revealed two primary issues affecting the stability and success rate of these tests:

**Timeout Duration**: Previously set at 20 minutes, this duration proved insufficient, leading to premature test terminations especially notable in tests that parallel those in the LSF submodules where the timeout is set at 45 minutes.

#### What does this fix?
This PR addresses the above issues by implementing the following changes:

**Increased Timeout**: The timeout for tests has been adjusted from 20 minutes to 45 minutes to align with the LSF submodule settings and accommodate tests that require longer execution times.
**Set build to production**: Set NODE_ENV to production and BUILD_NO_MINIMIZATION to false so it prevent the tests to fail; the reason: development mode loaded some task at the beginning, messing with the LSF instance in tests.

### What level of testing was included in the change?
_(check all that apply)_
- [x] e2e
- [ ] integration
- [ ] unit

### CODE REVIEW

**Comment by juliosgarbi** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8648660563)
> Successfully merged:  4 files changed, 52 insertions(+), 7 deletions(-)

**Comment by juliosgarbi** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8650039411)
> Successfully merged:  7 files changed, 37 insertions(+), 22 deletions(-)

**Comment by bmartel** [comment]
>Added Waits: Additional wait times have been introduced in the taxonomy tests to ensure **adequate time for location validation**. This change has been tested on Feature branches and has shown to prevent the previously encountered failures by allowing proper loading and processing times.

@juliosgarbi Do we have a more tactical means to identify if the taxonomy is loaded and ready? This could possibly introduce a flaky test if there is not a way to guarantee the test can proceed safely. Also what is meant by the bolded part? Does that mean there is some validation that happens on load? Is this asynchronously loaded from a remote endpoint, or is it something we have loaded and controlled from the local test instance?

**Comment by hlomzik** [comment]
Yea, I would also wait against random `wait`s with magic numbers, especially given that they added before LS init! Meaning that there is nothing on the page at this moment, so it waits for nothing. Orrrr... If there is some default task loaded every time we open LSF standalone, then we can replace this task to something lightweight.

File: web/libs/editor/tests/e2e/tests/taxonomy.test.js [hunk 1]
```diff
@@ -34,36 +34,35 @@ Scenario('Lines overlap', async ({ I, LabelStudio, AtTaxonomy }) => {
...
 
-
   I.amOnPage('/');
   LabelStudio.init({
     config: `
 <View>
   <Text name="text" value="$text"/>
   <Taxonomy name="taxonomy" toName="text">
-    <Choice value="target group">
+    <Choice value="target group 2">
```
Comment (Gondragos) [inline web/libs/editor/tests/e2e/tests/taxonomy.test.js]: The changes like that could potentially break the test logic. Target group title most probably should be the same in each config.

**Comment by juliosgarbi** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8707030530)
> Successfully merged:  328 files changed, 99 insertions(+), 16 deletions(-)

**Comment by juliosgarbi** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8724452622)
> Successfully merged:  7 files changed, 65 insertions(+), 10 deletions(-)

**Review by hlomzik — APPROVED** [review]

**Review by farioas — APPROVED** [review]

**Review by bmartel — APPROVED** [review]

**Comment by juliosgarbi** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8755025167)
> Successfully merged:  43 files changed, 774 insertions(+), 789 deletions(-)

**Pull request merged**